db/migrations/20230904154152_initial_war_migration.php

```diff
@@ -0,0 +1,198 @@
+<?php
+declare(strict_types=1);
+
+use Phinx\Migration\AbstractMigration;
+
+final class InitialWarMigration extends AbstractMigration
+{
+    /**
+     * Migrate Up.
+     */
+    public function up(): void
+    {
+        // Start Transaction
+        $this->execute("START TRANSACTION");
+
+        // Update region_locations - setting resource to ID value
+        $this->execute("UPDATE region_locations SET `resource` = 1 where `resource` = 'materials'");
+        $this->execute("UPDATE region_locations SET `resource` = 2 where `resource` = 'food'");
+        $this->execute("UPDATE region_locations SET `resource` = 3 where `resource` = 'wealth'");
+        $this->execute("ALTER TABLE `region_locations` CHANGE `resource` `resource_id` INT(11) NULL DEFAULT NULL");
+        $this->execute("UPDATE `region_locations` SET `resource_id` = 4 WHERE `type` = 'castle' AND `region_id` IN (6, 7, 8)");
+        $this->execute("UPDATE `region_locations` SET `resource_id` = 5 WHERE `type` = 'castle' AND `region_id` IN (9, 10, 11)");
+        $this->execute("UPDATE `region_locations` SET `resource_id` = 6 WHERE `type` = 'castle' AND `region_id` IN (12, 13, 14)");
+        $this->execute("UPDATE `region_locations` SET `resource_id` = 7 WHERE `type` = 'castle' AND `region_id` IN (15, 16, 17)");
+        $this->execute("UPDATE `region_locations` SET `resource_id` = 8 WHERE `type` = 'castle' AND `region_id` IN (18, 19, 20)");
+
+        // Update region_locations - add columns for production
+        $this->execute("ALTER TABLE `region_locations` ADD `resource_production` INT(11) DEFAULT 0");
+        $this->execute("ALTER TABLE `region_locations` ADD `resource_penalty` INT(11) DEFAULT 0");
+        $this->execute("ALTER TABLE `region_locations` ADD `defense` INT(11) NOT NULL DEFAULT '0' AFTER `resource_penalty`");
+        $this->execute("UPDATE `region_locations` SET `resource_production` = 25");
+        $this->execute("UPDATE `region_locations` SET `defense` = 75 where `type` = 'castle'");
+        $this->execute("UPDATE `region_locations` SET `defense` = 50 where `type` = 'village'");
+        $this->execute("UPDATE `region_locations` SET `name` = 'Village' where `type` = 'village'");
+        $this->execute("UPDATE `region_locations` SET `name` = 'Castle' where `type` = 'castle'");
+
+        // Update villages table - add column for resources
+        $this->execute("ALTER TABLE `villages` ADD `resources` VARCHAR(200) DEFAULT '[]'");
```

Comment:
```suggestion
        $this->execute("ALTER TABLE `villages` ADD `resources` VARCHAR(1000) DEFAULT '[]'");
```
What's the longest conceivable length of serialized resources you could see being in this column?

Reply:
8 resources with IDs, plus their value and wrapper
{"1":9999999,"2":9999999,"3":9999999,"4":9999999,"5":9999999,"6":9999999,"7":9999999,"8":9999999} <- 97 characters